src/hooks/use-on-focus-outside.js

```diff
@@ -3,15 +3,18 @@ import { useEffect } from 'react'
 /**
  * Hook that runs a callback on focus outside the ref
  */
-function useOnFocusOutside(ref, handler) {
+function useOnFocusOutside(ref, handler, shouldListen = true) {
   useEffect(() => {
     function handleFocusChange(event) {
       const isOutside = ref.current && !ref.current.contains(event.target)
       if (isOutside) handler(event)
     }
 
-    // Bind the event listener
-    document.addEventListener('focusin', handleFocusChange)
+    // Bind the event listener if the listener should be applied via `shouldListen`
+    if (shouldListen) {
+      document.addEventListener('focusin', handleFocusChange)
+    }
```

Comment:
Added `shouldListen` that works the same as `useOnClickOutside`'s `shouldListen`, that way it's possible to conditionally add this listeners instead of adding it all the time.